docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,493 @@
+
+**Pros**
+
+- Single infra cluster kind irrespective of if you are creating an unmanaged or GKE based cluster. It doesn’t require the user to pick the right one.
+- Clear separation between cluster infrastructure and the actual managed (i.e. GKE) control plane
+- Works with cluster class
+
+**Cons**
+
+- Additional complexity and logic in the infra cluster controller
+- API definition could be messy if only certain fields are required for one type of cluster
+
+## Recommendations
+
+It is proposed that option 3 (two kinds with a managed control plane and managed infra cluster with better separation of responsibilities) is the best way to proceed for **new implementations** of managed Kuberenetes in a provider.
+
+The reasons for this recommendation are as follows:
+
+- It adheres closely to the original separation of concerns between the infra and control plane providers
+- The infra cluster provisions and manages the general infrastructure required for the cluster but not the control plane.
+- By having a separate infra cluster API definition, it allows differences in the API between managed and unmanaged clusters.
+
+Providers like CAPZ and CAPA have already implemented managed Kubernetes support and there should be no requirement on them to move to Option 3. Both Options 2 and 4 are solutions that would work with ClusterClass and so could be used if required.
+
+Option 1 is the only option that will not work with ClusterClass and would require a change to CAPI. Therefore this option is not recommended.
+
+*** This means that CAPA will have to make changes to move away from Option 1 if it wants to support ClusterClass.
+
+### Additional notes on option 3
+
+There are a number of cons listed for option 3. With having 2 API kinds for the infra cluster (and associated controllers), there is a risk of code duplication. To reduce this the 2 controllers can utilize shared reconciliation code from the different controllers so as to reduce this duplication.
+
+The user will need to be aware of when to use which specific infra cluster kind. In our example this means that a user will need to know when to use `GCPCluster` vs `GCPManagedCluster`. To give clear guidance to users, we will provide templates (including ClusterClasses) and documentation for both the unmanaged and managed varieties of clusters. If we used the same infra cluster kind across both unmanaged & managed (i.e. option 4) then we run the risk of complicating the API for the infra cluster & controller if the required properties diverge.
+
+### Managed Node Groups for Worker Nodes
+
+Some cloud providers also offer Managed Node Groups as part of their Managed Kubernetes service as a way to provision worker nodes for a cluster. For example, in GCP there are Node Pools and in AWS there are EKS Managed Node Groups.
+
+There are 2 different ways to represent a group of machines in CAPI:
```

Comment:
this is inaccurately ruling out MachineSets which are a valid and supported use case.

Reply:
Oh, this is interesting to hear. 

When would people use MachineSets directly instead of MachineDeployments in CAPI? Asking because I didn't know that case existed, always thought people interact with MachineDeployments.

Reply:
Very generically speaking. I would assume folks who have their own MachineDeployment (or something similar) implementation